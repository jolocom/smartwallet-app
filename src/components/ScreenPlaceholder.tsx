import React from 'react'
import { View, StyleSheet } from 'react-native'

import JoloText, { JoloTextKind, JoloTextWeight } from '~/components/JoloText'
import { JoloTextSizes } from '~/utils/fonts'
import { Colors } from '~/utils/colors'

interface Props {
  title: string
  description: string
}

const ScreenPlaceholder: React.FC<Props> = ({
  title,
  description,
  children,
}) => {
  return (
    <View style={styles.container}>
      <View style={{ width: '100%' }}>
        <JoloText testID="title" kind={JoloTextKind.title}>
          {title}
        </JoloText>
        <JoloText
          testID="description"
          size={JoloTextSizes.mini}
          color={Colors.white50}
          weight={JoloTextWeight.regular}
          customStyles={{ marginVertical: 12, paddingHorizontal: 32 }}
        >
          {description}
        </JoloText>
      </View>
      {children}
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    height: '100%',
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
    paddingBottom: '20%',
  },
})

export default ScreenPlaceholder
