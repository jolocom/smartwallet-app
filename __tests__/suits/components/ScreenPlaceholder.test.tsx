import { render } from '@testing-library/react-native'
import React from 'react'
import { Platform } from 'react-native'
import ScreenPlaceholder from '~/components/ScreenPlaceholder'

describe('ScreenPlaceholder', () => {
  const defaultProps = {
    title: 'test title',
    description: 'test description',
  }

  it('should match snapshot', () => {
    const rendered = render(<ScreenPlaceholder {...defaultProps} />).toJSON()

    expect(rendered).toMatchSnapshot()
  })

  it('should render the title', () => {
    const rendered = render(<ScreenPlaceholder {...defaultProps} />)

    const title = rendered.getByTestId('title')
    expect(title.props.children).toBe(defaultProps.title)
  })

  it('should render the description', () => {
    const rendered = render(<ScreenPlaceholder {...defaultProps} />)

    const description = rendered.getByTestId('description')
    expect(description.props.children).toBe(defaultProps.description)
  })

  it('description should be of color Colors.white50', () => {
    const rendered = render(<ScreenPlaceholder {...defaultProps} />)

    const description = rendered.getByTestId('description')
    expect(description.props.style).toMatchObject([
      {
        textAlign: 'center',
        paddingTop: Platform.select({ ios: 5, android: 0 }),
      },
      {
        fontSize: 16,
        lineHeight: 18,
        letterSpacing: 0,
        color: 'rgba(255, 255, 255, 0.50)',
        fontFamily: 'TTCommons-Regular',
      },
      { marginTop: 12, paddingHorizontal: 32 },
    ])
  })
})
