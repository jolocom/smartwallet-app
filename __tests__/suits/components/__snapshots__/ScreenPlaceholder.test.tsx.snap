// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenPlaceholder should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "paddingBottom": "20%",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "paddingTop": undefined,
          "textAlign": "center",
        },
        Object {
          "color": "rgba(255,255,255,0.85)",
          "fontFamily": "TTCommons-Medium",
          "fontSize": 28,
          "letterSpacing": 0,
          "lineHeight": 32,
        },
        undefined,
      ]
    }
    testID="title"
  >
    test title
  </Text>
  <Text
    style={
      Array [
        Object {
          "paddingTop": undefined,
          "textAlign": "center",
        },
        Object {
          "color": "rgba(255, 255, 255, 0.50)",
          "fontFamily": "TTCommons-Regular",
          "fontSize": 16,
          "letterSpacing": 0,
          "lineHeight": 18,
        },
        Object {
          "marginTop": 12,
          "paddingHorizontal": 32,
        },
      ]
    }
    testID="description"
  >
    test description
  </Text>
</View>
`;
